### ISSUE TITLE
Change regex for interface down message on IOS

### ISSUE THREAD

**Issue by bittsi** [napalm-automation/napalm-logs#237]
Match not found for interface down message on CSR1kV:
`No OS matched for: <189>117: NetAuto_CSRv-03: May  9 12:49:27.098: %LINK-5-CHANGED: Interface GigabitEthernet2, changed state to administratively down`

There seems to be two issues:
1. No * before the date
2. Hostname consists -

My syslog message:
`<189>117: NetAuto_CSRv-03: May  9 12:49:27.098: %LINK-5-CHANGED: Interface GigabitEthernet2, changed state to administratively down`

 The syslog test message in the nampalm-logs project:
`<189>521: router1: *Nov 14 08:30:56.699: %LINK-5-CHANGED: Interface GigabitEthernet2, changed state to administratively down`

**No * before the date:**
Difference between the syslog messages:
`NetAuto_CSRv-03: May` vs. `router1: *Nov`

**Hostname consists -:**
Difference between the syslog messages:
`NetAuto_CSRv-03` vs. `router1`

**Suggestion to solve that:**
Add ? behind the * and allow non-alphanumeric characters:
`\<(\d+)\>(\d+)\:\s+([^ ]+)\:\s+\*?(\w+\s?\W?\d+)\s+(\d\d:\d\d:\d\d)\.(\d+)\:\s+\%([\w-]+)\:\s+(.*)`

**Comment by bittsi** [comment]
An asterisk (*) before the time indicates that the time may be incorrect because the system clock has not synchronized to a reliable time source.
[https://www.cisco.com/c/en/us/td/docs/ios-xml/ios/esm/configuration/xe-3s/esm-xe-3s-book/esm-syslog.html]

**Comment by loverend** [comment]
`IOS` and `IOS EX` look to be 2 distinct OS. To add support for `IOS EX` we should add a new OS under https://github.com/napalm-automation/napalm-logs/tree/develop/napalm_logs/config

**Comment by loverend** [comment]
@mirceaulinic just advised that there is quite a good overlap with these 2 os, so we can update with your new regex. Can you add a PR @bittsi ?

Thanks

**Comment by bittsi** [comment]
Will try to do that, but need some advice.
Can I add a second line to the syslog.msg](https://github.com/napalm-automation/napalm-logs/blob/develop/tests/config/ios/INTERFACE_DOWN/default/syslog.msg) for testing?

Then I have to adapt the [init file](https://github.com/napalm-automation/napalm-logs/blob/develop/napalm_logs/config/ios/init.yml):
Is it best to add another prefix for the `*?` part?

How can I test my changes locally?

**Comment by loverend** [comment]
To test locally you can do:

```
sudo <path to virtualenv>/bin/python napalm_logs/scripts/cli.py --disable-security --log-file cli -l debug
```

Then in a different terminal run the client:

```
<path to venv>/napalm-logs/bin/python examples/client_no_auth.py
```

Then send a log message to udp 514:

```
echo "<189>521: router1: *Nov 14 08:30:56.699: %LINK-5-CHANGED: Interface GigabitEthernet2, changed state to administratively down" > /dev/udp/0.0.0.0/514
```

To add a second test you can just add another directory along side `default`, here is an example for `nxos`:

https://github.com/napalm-automation/napalm-logs/tree/develop/tests/config/nxos/INTERFACE_DOWN

The only change we need to make to the regex is to move the `*` before `date` into the regex and make it optional:

```
diff --git a/napalm_logs/config/ios/init.yml b/napalm_logs/config/ios/init.yml
index bab7955..7032918 100644
--- a/napalm_logs/config/ios/init.yml
+++ b/napalm_logs/config/ios/init.yml
@@ -6,19 +6,19 @@ prefixes:
     values:
       messageId: (\d+)
       host: ([^ ]+)
-      date: (\w+\s?\d+)
+      date: \*?(\w+\s?\d+)
       time: (\d\d:\d\d:\d\d)
       milliseconds: (\d\d\d)
       tag: ([\w-]+)
       # some messages contain a process number (eg. OSPF related messages)
       processId: (\d+)
-    line: '{messageId}: {host}: *{date} {time}.{milliseconds}: %{tag}: Process {processId}, '
+    line: '{messageId}: {host}: {date} {time}.{milliseconds}: %{tag}: Process {processId}, '
   - time_format: "%b %d %H:%M:%S"
     values:
       messageId: (\d+)
       host: ([^ ]+)
-      date: (\w+\s?\d+)
+      date: \*?(\w+\s?\d+)
       time: (\d\d:\d\d:\d\d)
       milliseconds: (\d+)
       tag: ([\w-]+)
-    line: '{messageId}: {host}: *{date} {time}.{milliseconds}: %{tag}: '
+    line: '{messageId}: {host}: {date} {time}.{milliseconds}: %{tag}: '
```

The regex for `host` matches everything other than whitespace `([^ ]+)`

**Comment by bittsi** [comment]
@loverend Thanks for the explanations. Unfortunately I couldn't send a log message locally on mac. Will try to do it on a linux server next week.

**Comment by loverend** [comment]
I have just commended on the PR with the expected output. I hadn't seen this comment before I wrote that update.

To test on mac you should be able to send the message using `nc`:

```
nc -u localhost 514
```

Then paste the whole message.

**Comment by bittsi** [comment]
Solved with [87b73fc](https://github.com/napalm-automation/napalm-logs/commit/87b73fc495767adbc50628a51d4bb1215f2fc47d)
